rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/activityList.spec.ts

```diff
@@ -0,0 +1,363 @@
+        if (url.includes("running")) {
+          return Promise.resolve({
+            data: {
+              executions: mockRunningExecutions,
+            },
+          });
+        }
+        if (url.includes("eventsAjax")) {
+          return Promise.resolve({
+            data: {
+              total: 20,
+              offset: 0,
+              reports: reports,
+            },
+          });
+        }
+        return Promise.resolve({ data: {} });
+      });
+      const wrapper = await shallowMountActivityList();
+      await wrapper.vm.$nextTick(); // Wait for all pending updates to finish
+      // Verify the current executions
+      const executionItems = wrapper.findAll('[data-testid="execution-link"]');
+      expect(executionItems.length).toBe(2); // Directly using the expected length value
+      // Verify the finished job reports
+      const reportItems = wrapper.findAll('[data-testid="report-row-item"]');
+      expect(reportItems.length).toBe(2);
+    });
+
+    it("error message when loading(apis) fails", async () => {
+      axiosMock.get.mockRejectedValue(new Error("An error occurred"));
+      const wrapper = await shallowMountActivityList();
+      await wrapper.vm.$nextTick();
+      const errorMessage = wrapper.find('[data-testid="error-message"]');
+      expect(errorMessage.text()).toContain("An Error Occurred");
+    });
+
+    it("message when there is no info available", async () => {
+      axiosMock.get.mockImplementation((url) => {
+        if (url.includes("eventsAjax")) {
+          return Promise.resolve({
+            data: {
+              reports: [],
+              total: 0,
+              offset: 0,
+              lastDate: -1,
+            },
+          });
+        }
+        if (url.includes("running")) {
+          return Promise.resolve({
+            data: {
+              executions: [],
+            },
+          });
+        }
+        return Promise.resolve({ data: {} });
+      });
+      const wrapper = await shallowMountActivityList();
+      await wrapper.vm.$nextTick();
+      const noInfoMessage = wrapper.find('[data-testid="no-data-message"]');
+      expect(noInfoMessage.text()).toBe("No results for the query");
+    });
+  });
+
+  describe("trigger bulk actions", () => {
+    it("trigger bulk actions - delete", async () => {
+      axiosMock.get.mockImplementation((url) => {
+        if (url.includes("eventsAjax")) {
+          return Promise.resolve({
+            data: {
+              total: reports.length,
+              offset: 0,
+              reports: reports,
+            },
+          });
+        }
+        if (url.includes("running")) {
+          return Promise.resolve({
+            data: {
+              executions: mockRunningExecutions,
+            },
+          });
+        }
+        return Promise.resolve({ data: {} });
+      });
+      axiosMock.post.mockResolvedValueOnce({
+        data: { allsuccessful: true },
+      });
```

Comment:
Please remove this as it's not being used - the method executionBulkDelete isn't using axios